spec/tbd_tests_spec.rb

```diff
@@ -2121,6 +2123,58 @@
     # masonry shelf angles). Users are always free to curtomize TBD (via
     # JSON input) if needed.
 
+    # --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- #
+    # Test a custom non-0 "ceiling" PSI-factor.
+    file  = File.join(__dir__, "files/osms/out/z5.osm")
+    path  = OpenStudio::Path.new(file)
+    model = translator.loadModel(path)
+    expect(model).to_not be_empty
+    model = model.get
+
+    argh               = {}
+    argh[:option     ] = "uncompliant (Quebec)"
+    argh[:io_path    ] = File.join(__dir__, "../json/tbd_z5.json")
+    argh[:schema_path] = File.join(__dir__, "../tbd.schema.json")
+
+    json     = TBD.process(model, argh)
+    expect(json).to be_a(Hash)
+    expect(json).to have_key(:io)
+    expect(json).to have_key(:surfaces)
+    io       = json[:io      ]
+    surfaces = json[:surfaces]
+    expect(TBD.status).to be_zero
+    expect(TBD.logs).to be_empty
+    expect(surfaces).to be_a(Hash)
+    expect(surfaces.size).to eq(40)
+    expect(io).to be_a(Hash)
+    expect(io).to have_key(:edges)
+
+    # Plenum "walls" are (still) derated.
+    plnum_walls.each do |s|
+      expect(surfaces).to have_key(s)
+      expect(surfaces[s][:deratable]).to be true
+    end
+
+    # "Other" ceilings (i.e. those of conditioned spaces, adjacent to plenum
+    # "floors") are (still) no longer derated.
+    other_ceilings.each do |s|
+      expect(surfaces).to have_key(s)
+      expect(surfaces[s][:deratable]).to be false
+    end
+
+    io[:edges].select { |edge| edge[:type] == :ceiling }.each do |edge|
+      expect(edge[:psi]).to eq("salk")
+    end
+
+    expect(io[:edges].count { |edge| edge[:type] == :ceiling       }).to eq(4)
```

Comment:
The "Salk Institute"-like plenum floor edges remain tagged as "ceiling", yet retain a custom PSI-factor of 0.1 W/K per linear meter (not the default 0). Recourse to JSON customization implies additional steps in such cases, but "Salk Institute"-like plenums are a rare occurrence, albeit fascinating.